frontend/apps/marketing/src/components/button/ButtonContentfulDefinition.ts

```diff
@@ -0,0 +1,69 @@
+import {ComponentDefinition} from '@contentful/experiences-sdk-react';
+
+export const ButtonContentfulComponentDefinition: ComponentDefinition = {
+  id: 'button',
+  name: 'Button',
+  category: 'Custom Components',
+  thumbnailUrl:
+    'https://images.ctfassets.net/90t6bu6vlf76/6fRMP55wwDZF2C4ubzygTO/bad1643a5db519e1e3f6886f0f7bc7cd/component_button_thumbnail.png',
+  tooltip: {
+    description:
+      'Use a button to create clear calls to action. Supports different styles, sizes, and links to guide users to key actions.',
+    imageUrl:
+      'https://images.ctfassets.net/90t6bu6vlf76/6B9UVqsphQR9MUnNf4tYHU/94e6f32fcb793401f02399ceb0069722/component_button_tooltip.png',
+  },
+  variables: {
+    color: {
+      displayName: 'Color',
+      type: 'Text',
+      defaultValue: 'purple',
+      group: 'style',
+      validations: {
+        in: [
+          {value: 'purple', displayName: 'Purple'},
+          {value: 'black', displayName: 'Black'},
+          {value: 'white', displayName: 'White'},
+        ],
+      },
+    },
+    type: {
+      displayName: 'Type',
+      type: 'Text',
+      defaultValue: 'primary',
+      group: 'style',
+      validations: {
+        in: [
+          {value: 'primary', displayName: 'Primary'},
+          {value: 'secondary', displayName: 'Secondary'},
+        ],
+      },
+    },
+    text: {
+      displayName: 'Text',
+      type: 'Text',
+      defaultValue: 'Button',
+      group: 'content',
+    },
+    href: {
+      displayName: 'Link URL',
+      type: 'Text',
+      defaultValue: 'https://code.org',
+      group: 'content',
+    },
+    isLinkExternal: {
+      displayName:
+        'Is this link external? (Does this link leave the code.org site?)',
+      description:
```

Comment:
I added this description according to [documentations](https://www.contentful.com/developers/docs/experiences/component-definition-schema/#variable-definition-fields), but I'm not seeing it anywhere in contentfful. If someone will be able to make this appear somewhere in the contentful - that would be awesome. For now - will just leave it here

Reply:
I might have missed the discussion but did we also discuss programatically detecting whether a link is external or not? My suspicion is that prebinding will not automatically check this box and this will probably end up being unchecked most of the time even though the link is external. 

We could use something like `window.origin.hostname` to detect whether we are going external.

For the purposes of this PR, we can move external link target detection functionality to a separate task. My strong preference is to remove the need for content editors to make the determination of whether a link is external or not as these are most likely going to be prebound via content records and not individually updated very often.

Reply:
I also added this help text in the Content Model, so we should hopefully get another layer of "validation" here:

<img width="933" alt="Screenshot 2025-02-07 at 8 34 10 AM" src="https://github.com/user-attachments/assets/75a8bcf7-56ea-4c8b-bf7e-03bb32d21823" />

Reply:
To answer your original question, I don't know where these show up either 🤔 You could ask the Contentful folks in the contentful-codeorg Slack channel.

Reply:
That makes sense, could we remove the external link variable in this PR and have the external link logic as a separate Jira/PR?

It looks good otherwise!

Reply:
I don't think we should remove that checkbox now